### DISCUSSION TITLE
CMake Usage flag for find_package commands

### DISCUSSION THREAD

**Discussion by jackadrianglass** [microsoft/vcpkg#25383] [Ideas]
Currently when I want to figure out how to include a library into my cmake build file, I do the following command
`vcpkg install <port>`

This gives me the usage instructions for the `find_package` command and the `target_link_libraries`. What I'd like to be able to do is the following command:
`vcpkg usage <port>`

This would skip all the checks for whether or not the package needs to be installed and just print the usage strings for the particular port.

An extension to this would be if you typed `vcpkg usage` in a directory that has a manifest file in it, then it would print out all the usage strings for the dependencies listed in the manifest file.

A super extension would be to be able to write out these usage instructions to a simple text file for future reference.
`vcpkg usage -o usage.txt`

**Comment by pratikpc** [comment]
Adding to this, I would also like if this feature worked well with Boost

Recently the Boost behaviour changed.

I have a package called GoldPKG which simply scrapes the vcpkg install find_* and target_* commands and adds them to CMake files and something like this would definetely be beneficial for my use case.

**Comment by vicroms** [comment]
For now I have created PR #15243 which partially implements the features requested here. 
Unfortunately, I have other priorities to work on at the moment, so I won't be able to implement anything beyond what's already on the PR.

I'm counting on the community to implement the rest of the features in subsequent PRs  😉 

At @pratikpc what issues have you found with Boost?

**Comment by pratikpc** [comment]
The output for Boost now is this

    find_package(Boost REQUIRED [COMPONENTS <libs>...])
    target_link_libraries(main PRIVATE Boost::boost Boost::<lib1> Boost::<lib2> ...)

If we could instead have something relevant to the Boost package at hand

For example, with Boost Array,

```
target_include_directories(main PRIVATE ${BOOST_ARRAY_INCLUDE_DIRS})
```

would be better output
